docs/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-6.0/enable-consistent-topology.rst

```diff
@@ -3,4 +3,102 @@ Enable Consistent Topology Updates
+In those clusters, consistent topology changes need to be enabled with a dedicated upgrade procedure.
+
+Before running the procedure, you **must** check that the cluster meets some prerequisites
+and you **must** ensure that some administrative procedures will not be run
+while the upgrade procedure is in progress.
+
+.. _enable-raft-topology-6.0-prerequisites:
+
+Prerequisites
+=============
+
+* Make sure that all nodes in the cluster are upgraded to ScyllaDB Open Source 6.0.
+* Verify that :ref:`schema on raft is enabled <schema-on-raft-enabled>`.
+* Make sure that all nodes enabled ``SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES`` cluster feature.
+  One way to verify this is to look for the following message in the log:
+
+  .. code-block:: none
+
+    features - Feature SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES is enabled
+
+  Alternatively, this can be checked programmatically by checking whether ``value`` column under the key ``enabled_features`` contains the name of the feature in the ``system.scylla_local`` table. This can be done with the following query:
+
+  .. code-block:: none
+
+    select value from system.scylla_local where key = 'enabled_features';
+
+  It returns a string which contains a comma-separated list of names of the features that the node sees as enabled. You should wait until it contains ``SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES``.
+
+* Make sure that all nodes are alive for the duration of the upgrade.
+
+.. _enable-raft-topology-6.0-forbidden-operations:
+
+Administrative operations which must not be running during upgrade
+==================================================================
+
+Make sure that administrative operations will not be running while upgrade is in progress.
+In particular, you must abstain from:
+
+* :doc:`Cluster management procedures </operating-scylla/procedures/cluster-management/index>` (adding, replacing, removing, decomissioning nodes etc.).
+* Runing :doc:`nodetool repair </operating-scylla/nodetool-commands/repair>`.
+* Runing :doc:`nodetool checkAndRepairCdcStreams </operating-scylla/nodetool-commands/checkandrepaircdcstreams>`.
+
+Running the procedure
+=====================
+
+.. warning::
+
+  Before proceeding, make sure that all the :ref:`prerequisites <enable-raft-topology-6.0-prerequisites>` are met
+  and no :ref:`forbidden administrative operations <enable-raft-topology-6.0-forbidden-operations>` will run
+  during upgrade. Failing to do so may put the cluster in an inconsistent state.
+
+Starting the upgrade procedure is done by issuing an HTTP request to any of the nodes in the cluster.
+
+For example, you can start the upgrade using the ``curl`` command like this:
+
+.. code-block:: bash
+
+	curl -X POST "http://127.0.0.1:10000/storage_service/raft_topology/upgrade"
+
+Next, wait until all nodes report that upgrade is complete. Checking that a single node finished upgrade can be done in a number of ways:
```

Comment:
Too many options, instructions unclear, ~~p~~hand stuck in toaster

Maybe let's distinguish one of the options as recommended one, and the other options as optional?

Reply:
While looking at the logs is probably the least reliable way to do it and I can drop it, I'm not sure about whether CQL or HTTP is better.

Reply:
I removed the log-gazing method and left CQL + HTTP as two equally valid options. Let me know if you would like me to promote one over the other.